dev/Interop/StoragePickers/PickerLocalization.cpp

```diff
@@ -9,14 +9,119 @@
 #include "PickerLocalization.h"
 #include <winrt\base.h>
 #include <winrt\Microsoft.Windows.ApplicationModel.Resources.h>
-#include <iostream>
 
+// find framework package related code from https://github.com/microsoft/microsoft-ui-xaml/blob/main/src/controls/dev/ResourceHelper/ResourceAccessor.cpp
+// and https://github.com/microsoft/microsoft-ui-xaml/blob/main/src/controls/dev/dll/SharedHelpers.cpp
+
+namespace {
+    constexpr wchar_t c_reunionPackageNamePrefix[] = L"Microsoft.WindowsAppRuntime";
+    constexpr int c_reunionPackageNamePrefixLength = ARRAYSIZE(c_reunionPackageNamePrefix) - 1;
+
+    // Tries to retrieve the current package graph. Will return true if a package graph exists
+    // and is non-empty, else false.
+    bool TryGetCurrentPackageGraph(
+        const std::uint32_t flags,
+        std::uint32_t& packageCount,
+        const PACKAGE_INFO*& packageGraph,
+        std::unique_ptr<BYTE[]>& buffer)
+    {
+        std::uint32_t bufferLength{};
+        LONG rc{ ::GetCurrentPackageInfo(flags, &bufferLength, nullptr, nullptr) };
+        if ((rc == APPMODEL_ERROR_NO_PACKAGE) || (rc == ERROR_SUCCESS))
+        {
+            // No/empty package graph
+            return false;
+        }
+        else if (rc != ERROR_INSUFFICIENT_BUFFER)
+        {
+            if (FAILED(HRESULT_FROM_WIN32(rc)))
+            {
+                return false;
+            }
+        }
+
+        buffer = std::make_unique<BYTE[]>(bufferLength);
+        if (SUCCEEDED(HRESULT_FROM_WIN32(::GetCurrentPackageInfo(flags, &bufferLength, buffer.get(), &packageCount))))
+        {
+            packageGraph = reinterpret_cast<PACKAGE_INFO*>(buffer.get());
+            return true;
+        }
+
+        return false;
+    }
+
+    bool IsInFrameworkPackageImpl(winrt::hstring& frameworkInstallLocation)
+    {
+        // We could be running in an app that is using WinUI "In-App" (i.e. the WinUI dlls are deployed as part of the app). 
```

Comment:
```suggestion
        // We could be running in an app that is using WinUI Self-Contained (i.e. the WinUI dlls are deployed as part of the app). 
```

Reply:
Also: Change `WinUI` to `WinAppSDK` (in both references)